### PR TITLE
Add support files to get passkey credential and passkey assertion

### PR DESCRIPTION
**Pull request by juan-arias**
## Proposed changes

Add support files to get passkey credential and passkey assertion

## Type of change

- [x] Feature work
- [ ] Bug fix
- [ ] Documentation
- [ ] Engineering change
- [ ] Test
- [ ] Logging/Telemetry

## Risk

- [ ] High – Errors could cause MAJOR regression of many scenarios. (Example: new large features or high level infrastructure changes)
- [ ] Medium – Errors could cause regression of 1 or more scenarios. (Example: somewhat complex bug fixes, small new features)
- [x] Small – No issues are expected. (Example: Very small bug fixes, string changes, or configuration settings changes)

### CODE REVIEW

File: IdentityCore/src/broker_operation/request/MSIDBrokerOperationPasskeyAssertionRequest.h [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+// THE SOFTWARE.  
+
+
+#import "MSIDBrokerOperationRequest.h"
+
+@class MSIDAccountIdentifier;
+
+NS_ASSUME_NONNULL_BEGIN
+
+@interface MSIDBrokerOperationPasskeyAssertionRequest : MSIDBrokerOperationRequest
```
Comment (oldalton) [inline IdentityCore/src/broker_operation/request/MSIDBrokerOperationPasskeyAssertionRequest.h:32]: Can you wrap it in macros that exclude it from MSAL CPP?

File: IdentityCore/src/broker_operation/request/MSIDBrokerOperationPasskeyCredentialRequest.m [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+- (instancetype)initWithJSONDictionary:(NSDictionary *)json error:(NSError **)error
+{
+    self = [super initWithJSONDictionary:json error:error];
+
+    return self;
+}
+
+- (NSDictionary *)jsonDictionary
+{
+    NSMutableDictionary *json = [[super jsonDictionary] mutableCopy];
```
Comment (oldalton) [inline IdentityCore/src/broker_operation/request/MSIDBrokerOperationPasskeyCredentialRequest.m:55]: I think mutableCopy here is unnecessary

File: IdentityCore/src/broker_operation/response/MSIDBrokerOperationGetPasskeyAssertionResponse.m [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+
+- (instancetype)initWithJSONDictionary:(NSDictionary *)json error:(__unused NSError **)error
+{
+    self = [super initWithJSONDictionary:json error:error];
+
+    if (self)
+    {
+        if (self.success)
+        {
+            _passkeyAssertion =[[MSIDPasskeyAssertion alloc] initWithJSONDictionary:json error:nil];
```
Comment (oldalton) [inline IdentityCore/src/broker_operation/response/MSIDBrokerOperationGetPasskeyAssertionResponse.m:53]: Should we capture and log the error?

File: IdentityCore/src/broker_operation/response/MSIDBrokerOperationGetPasskeyCredentialResponse.m [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+
+- (instancetype)initWithJSONDictionary:(NSDictionary *)json error:(__unused NSError **)error
+{
+    self = [super initWithJSONDictionary:json error:error];
+
+    if (self)
+    {
+        if (self.success)
+        {
+            _passkeyCredential =[[MSIDPasskeyCredential alloc] initWithJSONDictionary:json error:nil];
```
Comment (oldalton) [inline IdentityCore/src/broker_operation/response/MSIDBrokerOperationGetPasskeyCredentialResponse.m:53]: Same here, should the error get captured?

File: IdentityCore/src/broker_operation/response/MSIDPasskeyAssertion.m [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+
+- (instancetype)initWithJSONDictionary:(NSDictionary *)json error:(__unused NSError **)error
+{
+    self = [super init];
+
+    if (self)
+    {
+        _signature = [[json msidStringObjectForKey:MSID_BROKER_PASSKEY_ASSERTION_SIGNATURE_JSON_KEY] msidHexData];
+        _authenticatorData = [[json msidStringObjectForKey:MSID_BROKER_PASSKEY_ASSERTION_AUTHENTICATOR_DATA_JSON_KEY] msidHexData];
+        _credentialId = [[json msidStringObjectForKey:MSID_BROKER_PASSKEY_ASSERTION_CREDENTIAL_ID_JSON_KEY] msidHexData];
```
Comment (oldalton) [inline IdentityCore/src/broker_operation/response/MSIDPasskeyAssertion.m:63]: Should we check that all those present and fail +log if those aren't?

File: IdentityCore/src/util/NSString+MSIDExtensions.m [hunk 6]
```diff
@@ -276,6 +276,25 @@ - (BOOL)msidIsEquivalentWithAnyAlias:(NSArray<NSString *> *)aliases
     return result;
 }
 
+- (NSData *)msidHexData
```
Comment (oldalton) [inline IdentityCore/src/util/NSString+MSIDExtensions.m:279]: Can you add unit tests for this?

**Review by oldalton — COMMENTED** [review]
Can you wrap all those new classes in macros that exclude them from MSAL CPP code?

File: IdentityCore/src/broker_operation/request/MSIDBrokerOperationPasskeyAssertionRequest.h (hunk 1 above)
Comment (juan-arias) [reply IdentityCore/src/broker_operation/request/MSIDBrokerOperationPasskeyAssertionRequest.h:33]: Updated

File: IdentityCore/src/broker_operation/request/MSIDBrokerOperationPasskeyCredentialRequest.m (hunk 2 above)
Comment (juan-arias) [reply IdentityCore/src/broker_operation/request/MSIDBrokerOperationPasskeyCredentialRequest.m]: Right, updated.

File: IdentityCore/src/broker_operation/response/MSIDBrokerOperationGetPasskeyAssertionResponse.m (hunk 3 above)
Comment (juan-arias) [reply IdentityCore/src/broker_operation/response/MSIDBrokerOperationGetPasskeyAssertionResponse.m]: Updated

File: IdentityCore/src/broker_operation/response/MSIDBrokerOperationGetPasskeyCredentialResponse.m (hunk 4 above)
Comment (juan-arias) [reply IdentityCore/src/broker_operation/response/MSIDBrokerOperationGetPasskeyCredentialResponse.m]: Updated

File: IdentityCore/src/broker_operation/response/MSIDPasskeyAssertion.m (hunk 5 above)
Comment (juan-arias) [reply IdentityCore/src/broker_operation/response/MSIDPasskeyAssertion.m:68]: Updated

File: IdentityCore/src/util/NSString+MSIDExtensions.m (hunk 6 above)
Comment (juan-arias) [reply IdentityCore/src/util/NSString+MSIDExtensions.m:279]: Added.

**Review by Veena11 — APPROVED** [review]

**Pull request merged**